src/jit/morph.cpp

```diff
@@ -3990,9 +4029,9 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* callNode)
     // Note that we do this for UNIX_AMD64 when we have a struct argument
     //
     if (!lateArgsComputed && (call->fgArgInfo->HasRegArgs()
-#ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
-                              || hasStructArgument
-#endif // FEATURE_UNIX_AMD64_STRUCT_PASSING
+ #ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
+                              || hasStackArgCopy
+ #endif // FEATURE_UNIX_AMD64_STRUCT_PASSING
```

Comment:
You should update the comment above this related to UNIX_AMD64 to be more clear about what the current situation is. In addition, it would be useful to state *why* we need to do this for UNIX_AMD64 structs.